app/src/main/java/com/gathering/android/event/putevent/pic/AddPicScreen.kt

```diff
@@ -222,49 +224,13 @@ class AddPicScreen : BottomSheetDialogFragment(), AddPicNavigator {
         Column(
             modifier = Modifier.fillMaxWidth(), horizontalAlignment = Alignment.CenterHorizontally
         ) {
-            Box {
-                IconButton(
-                    onClick = {
-                        if (imageUri != null) {
-                            onRotateClick()
-                        }
-                    }, modifier = Modifier.size(80.dp)
-                ) {
-                    Icon(
-                        imageVector = Icons.Default.RotateRight,
-                        contentDescription = null,
-                        modifier = Modifier.size(36.dp)
-                    )
-                }
-                ImageView(bmp = imageUri, size = 200.dp)
-            }
-            Row(
-                modifier = Modifier
-                    .fillMaxWidth()
-                    .padding(10.dp),
-                horizontalArrangement = Arrangement.Center
-            ) {
-                Button(onClick = {
-                    if (cameraPermissionState.status.isGranted) {
-                        onCameraClick()
-                    } else {
-                        cameraPermissionState.launchPermissionRequest()
-                    }
-                }) {
-                    Text(text = "OPEN CAMERA")
-                }
-                Spacer(modifier = Modifier.padding(10.dp))
-
-                Button(onClick = {
-                    if (galleryPermissionState.status.isGranted) {
-                        onGalleryClick()
-                    } else {
-                        galleryPermissionState.launchPermissionRequest()
-                    }
-                }) {
-                    Text(text = "OPEN GALLERY")
-                }
-            }
+            RotateIconAndImageView(imageUri, onRotateClick)
+            CameraAndGalleryButtons(
```

Comment:
i dont think grouping compostables like this makes sense. each composable should be a meaningful component. the fact that you couldn't find a good for it could be one reason its not a good candidate to be a component.

Reply:
done!